sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -13850,4 +13850,85 @@ public void testSurfaceErrorsWhenInsertingThroughIncorrectSelectStatment()
         "Possible error: SQL requires 'UNION' but only 'UNION ALL' is supported."
     );
   }
+
+  @Test
+  public void testInFilterLessThanInSubQueryThreshold() throws Exception
+  {
+    String query = "SELECT l1 FROM numfoo WHERE l1 IN (4842, 4844, 4845, 14905, 4853, 29064)";
+
+    testQuery(
+            PLANNER_CONFIG_DEFAULT,
+            QUERY_CONTEXT_DEFAULT,
+            DEFAULT_PARAMETERS,
+            query,
+            CalciteTests.REGULAR_USER_AUTH_RESULT,
+            ImmutableList.of(
+                    Druids.newScanQueryBuilder()
+                            .dataSource(CalciteTests.DATASOURCE3)
+                            .columns("l1")
+                            .intervals(querySegmentSpec(Filtration.eternity()))
+                            .context(QUERY_CONTEXT_DEFAULT)
+                            .legacy(false)
+                            .filters(in("l1",
+                                    ImmutableList.of(
+                                            "4842", "4844", "4845", "14905", "4853", "29064"),
+                                    null))
+                            .resultFormat(ResultFormat.RESULT_FORMAT_COMPACTED_LIST)
+                            .build()
+            ),
+            (sql, result) -> {
+              // Ignore the results, only need to check that the type of query is a filter.
+            },
+            null
+    );
+  }
+
+  @Test
+  public void testInFilterMoreThanInSubQueryThreshold() throws Exception
```

Comment:
1. this can be moved to `CalciteJoinQueryTest`.
2. please fix the closing brackets' indentation in multiline args - generally in the codebase then end on the same level as the starting of the user.
3. since we're only testing the plan, the test can be called like `testPlanWithInFilterMoreThanInSubQueryThreshold`

Reply:
Addressed